src/components/file-handler/FileHandler.jsx

```diff
@@ -118,37 +135,67 @@ const FileHandler = ({
     className
   );
 
-  const clickProps =
-    thumbnailSrc !== undefined && onClick
-      ? {onClick}
-      : {
-          href: src,
-          target: '_blank',
-          rel: 'noopener noreferrer',
-        };
+  const preventedOnClick =
+    onClick &&
+    (e => {
+      e.preventDefault();
+      return onClick();
+    });
+
+  const isActionProvided = src !== undefined || preventedOnClick;
 
+  const buttonEventProps = {
+    onClick: preventedOnClick,
+    onKeyDown: e => {
+      if (e.keyCode === 32 || e.keyCode === 13) {
+        return preventedOnClick && preventedOnClick(e);
+      }
+    },
+  };
+
+  const clickProps = preventedOnClick
+    ? buttonEventProps
+    : {
+        href: src,
+        target: '_blank',
+        rel: 'noopener noreferrer',
+      };
+
+  const role = clickProps.onClick && 'button';
   const thumbnail =
     thumbnailSrc !== undefined ? (
-      <img
-        {...clickProps}
-        src={thumbnailSrc}
-        alt=""
-        className="cursor-pointer"
-      />
+      <img src={thumbnailSrc} alt="" className="cursor-pointer" />
     ) : (
-      <a {...clickProps}>
-        <Icon type={iconType} size={24} color={ICON_COLOR['icon-black']} />
-      </a>
+      <Icon type={iconType} size={24} color={ICON_COLOR['icon-black']} />
```

Comment:
I think we can use just `color='icon-black'` here

Reply:
ok 👍🏻